src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/DiscoverVariantsFromContigAlignmentsSAMSpark.java

```diff
@@ -167,21 +180,119 @@ public static AlignedContig parseReadsAndOptionallySplitGappedAlignments(final I
+
+        final List<VariantContext> impreciseVariants =
+                Utils.stream(evidenceTargetLinks)
+                        .filter(DiscoverVariantsFromContigAlignmentsSAMSpark::isImpreciseDeletion)
+                        .filter(e -> e._2.getReadPairs() + e._2.getSplitReads() > parameters.impreciseEvidenceVariantCallingThreshold)
+                        .map(e -> createImpreciseDeletionVariant(e._2, reference.getReferenceSequenceDictionary(null), reference))
+                        .collect(Collectors.toList());
 
-        SVVCFWriter.writeVCF(pipelineOptions, vcfFileName, fastaReference, annotatedVariants, toolLogger);
+        localLogger.info("Called " + impreciseVariants.size() + " imprecise deletion variants");
+        collectedAnnotatedVariants.addAll(impreciseVariants);
+        return collectedAnnotatedVariants;
+    }
+
+    private static VariantContext createImpreciseDeletionVariant(final EvidenceTargetLink e,
+                                                                 final SAMSequenceDictionary sequenceDictionary,
+                                                                 final ReferenceMultiSource reference) {
+        final SvType svType = new SimpleSVType.ImpreciseDeletion(e, sequenceDictionary);
+        return AnnotatedVariantProducer
+                .produceAnnotatedVcFromEvidenceTargetLink(e, svType, sequenceDictionary, reference);
+    }
+
+    private static VariantContext annotateWithImpreciseEvidenceLinks(final VariantContext variant,
+                                                                     final PairedStrandedIntervalTree<EvidenceTargetLink> evidenceTargetLinks,
+                                                                     final SAMSequenceDictionary referenceSequenceDictionary,
+                                                                     final ReadMetadata metadata,
+                                                                     final int defaultUncertainty) {
+        if (variant.getStructuralVariantType() == StructuralVariantType.DEL) {
+            SVContext svc = SVContext.of(variant);
+            final int padding = (metadata == null) ? defaultUncertainty : (metadata.getMaxMedianFragmentSize() / 2);
+            PairedStrandedIntervals svcIntervals = svc.getPairedStrandedIntervals(referenceSequenceDictionary, padding);
+
+            final Iterator<Tuple2<PairedStrandedIntervals, EvidenceTargetLink>> overlappers = evidenceTargetLinks.overlappers(svcIntervals);
+            int readPairs = 0;
+            int splitReads = 0;
+            while (overlappers.hasNext()) {
+                final Tuple2<PairedStrandedIntervals, EvidenceTargetLink> next = overlappers.next();
+                readPairs += next._2.getReadPairs();
+                splitReads += next._2.getSplitReads();
+                overlappers.remove();
+            }
+            if (readPairs == 0 && splitReads == 0) {
```

Comment:
It feels more natural to me that these two annotations should exist for all (DEL for now) variants.

Reply:
OK. Removed this check. It feels like it might be more efficient to keep the check and maybe initialize all of the variants contexts to have these set to zero and not have to rebuild them at this step, but that seems less clean. What do you think?

Reply:
I like this idea. 
The change is most likely to happen in SVContext, with two over-loaded ctors, one with taking arguments of read pairs and split reads, one without and delegating to the other with default values. Would this work?

Reply:
Since I'm annotating preexisting variants and don't currently have the evidence target links at VC creation time, it's hard to do things in constructors without greatly changing the code path (such that I would have to broadcast out all of the evidence target links so that they would be there at VC creation). I'm not sure it's worth it to make that change now.